src/helper-functions/cart-helpers.js

```diff
@@ -0,0 +1,117 @@
+import axios from "axios";
+
+// A helper function to Add items in Cart
+const addToCart = async (productObj, boolFunc, notify) => {
+
+  try {
+    // token required as an authorization header
+    const encodedToken = localStorage.getItem("token")
+
+    await axios({
+      method: "POST",
+      headers: {
+        authorization: encodedToken,
+      },
+      url: "/api/user/cart",
+      data: JSON.stringify({
+        product: productObj,
+      }),
+    });
+
+    boolFunc();
+    notify();
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// A helper function to delete items from cart
+const deleteCartItem = async (id, dispatch, boolFunc) => {
+
+  try {
+    // token required as an authorization header
+    const encodedToken = localStorage.getItem("token")
+    const response = await axios({
+      method: "DELETE",
+      headers: {
+        authorization: encodedToken,
+      },
+      url: `/api/user/cart/${id}`,
+    });
+
+    boolFunc();
+    dispatch({ type: "SET_CART_DATA", payload: response.data.cart });
+  } catch (error) {
+    console.log(error)
+  }
+
+};
+
+// A helper function to increment cart item's quantity
+const incrementCartItem = async (id, dispatch, boolFunc) => {
+
+  try {
+    // token required as an authorization header
+    const encodedToken = localStorage.getItem("token")
+    const response = await axios({
+      method: "POST",
+      headers: {
+        authorization: encodedToken,
+      },
+      data: {
+        action: {
+          type: "increment",
+        },
+      },
+      url: `/api/user/cart/${id}`,
+    });
+
+    boolFunc();
+    dispatch({ type: "SET_CART_DATA", payload: response.data.cart });
+  }
+  catch (error) {
+    console.log(error)
+  }
+};
+
+// A helper function to decrement cart item's quantity
+const decrementCartItem = async (id, dispatch, boolFunc) => {
+
+  try {
+    // token required as an authorization header
+    const encodedToken = localStorage.getItem("token")
```

Comment:
instead of asking encoded token inside a function you can store inside a auth context